applications/StructuralMechanicsApplication/custom_conditions/point_moment_condition_3d.cpp

```diff
@@ -0,0 +1,243 @@
+        rValues[1] = r_angular_acc[1];
+        rValues[2] = r_angular_acc[2];
+    }
+    
+    //***********************************************************************
+    //***********************************************************************
+
+    void PointMomentCondition3D::CalculateAll( 
+        MatrixType& rLeftHandSideMatrix, 
+        VectorType& rRightHandSideVector,
+        ProcessInfo& rCurrentProcessInfo,
+        bool CalculateStiffnessMatrixFlag,
+        bool CalculateResidualVectorFlag 
+        )
+    {
+        KRATOS_TRY
+        
+        const unsigned int dim = 3;
+
+        // Resizing as needed the LHS
+        const unsigned int mat_size = dim;
+
+        if ( CalculateStiffnessMatrixFlag == true ) //calculation of the matrix is required
+        {
+            if ( rLeftHandSideMatrix.size1() != mat_size )
+            {
+                rLeftHandSideMatrix.resize( mat_size, mat_size, false );
+            }
+
+            noalias( rLeftHandSideMatrix ) = ZeroMatrix( mat_size, mat_size ); //resetting LHS
+        }
+
+        //resizing as needed the RHS
+        if ( CalculateResidualVectorFlag == true ) //calculation of the matrix is required
+        {
+            if ( rRightHandSideVector.size( ) != mat_size )
+            {
+                rRightHandSideVector.resize( mat_size, false );
+            }
+
+            noalias( rRightHandSideVector ) = ZeroVector( mat_size ); //resetting RHS
+        }
+
+        // Vector with a loading applied to the condition
+        array_1d<double, 3 > point_moment = ZeroVector(3);
+        if( this->Has( POINT_MOMENT ) )
+        {
+            noalias(point_moment) = this->GetValue( POINT_MOMENT );
+        }
+        
+        if( GetGeometry()[0].SolutionStepsDataHas( POINT_MOMENT ) )
+        {
+            noalias(point_moment) += GetGeometry()[0].FastGetSolutionStepValue( POINT_MOMENT );
+        }
+        
+        for(unsigned int k = 0; k < dim; ++k)
+        {
+            rRightHandSideVector[k] += GetPointMomentIntegrationWeight() * point_moment[k];
+        }
+
+        KRATOS_CATCH( "" )
+    }
+    
+    //************************************************************************************
+    //************************************************************************************
+    
+    double PointMomentCondition3D::GetPointMomentIntegrationWeight()
+    {
+        return 1.0;
+    }
+    
+    //***********************************************************************
+    //***********************************************************************
+    
+    int PointMomentCondition3D::Check( const ProcessInfo& rCurrentProcessInfo )
+    {
+        if ( ROTATION.Key() == 0 )
+        {
+            KRATOS_ERROR <<  "ROTATION has Key zero! (check if the application is correctly registered" << std::endl;
+        }
+
+        //verify that the dofs exist
+        for ( unsigned int i = 0; i < this->GetGeometry().size(); i++ )
+        {
```

Comment:
To reduce the this->GetGeometry() you can declare 
~~~
const GeometryType& this_geometry =this->GetGeometry();
~~~
This will reduce the number of calls (in this case maybe it does the same, what do you say @RiccardoRossi ?)